### DISCUSSION TITLE
PostgreSQL Error - directus database migrate:latest throwing error

### DISCUSSION THREAD

**Discussion by awnish-npg** [directus/directus#7285] [Archive]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues) that already describes my problem.

### Describe the Bug

I am using PostgreSQL for my application. I was using rc85 and now I have upgraded directus to rc88. So as it was suggested while running the "directus start", when I am running this command "directus database migrate:latest"
then I am getting error. I tried this after deleting the directus_setting table but still problem is not resolved.

![image](https://user-images.githubusercontent.com/60690093/128750559-60eb37a0-6967-4b10-abdd-26ef5ce2c61a.png)

### To Reproduce

running this command "directus database migrate:latest"

### What version of Directus are you using?

v9.0.0-rc.88

### What version of Node.js are you using?

14.16

### What database are you using?

Postgres 13

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying Directus?

CLI

**Comment by rijkvanzanten** [comment]
> when I am running this command "directus database migrate:latest" then I am getting error

What was the error you were getting originally?

> I tried this after deleting the directus_setting table but still problem is not resolved.

Sounds like you deleted one (or more) of the system tables Directus expects there to be. That'll definitely cause issues in running Directus and/or migrating to newer database versions..

**Reply by awnish-npg** [reply]
Let me explain the steps here.
1) I was having my DB in MySQL earlier and now I am migrating the DB to PostgreSQL
2) I used a tool "MySQL-to-PostgreSQL" using this tool https://www.convert-in.com/mysql-to-postgres.htm
3) After running the command "npx directus start" I get following message
![image](https://user-images.githubusercontent.com/60690093/128757410-5e741c6b-1ac6-4917-9877-30b30099f696.png)
4) So I executed the command "npx directus database migrate:latest" and got the error
![image](https://user-images.githubusercontent.com/60690093/128757570-23c1cddf-9dff-4f05-8e95-7d2330c583bd.png)
5) So I deleted the "storage_default_folder" column from the directus_settings and executed the database migrate:latest command and now getting this error
![image](https://user-images.githubusercontent.com/60690093/128757356-cdeebb36-8b81-4983-bbd3-ebd870dd9c77.png)

**Reply by rijkvanzanten** [reply]
> I used a tool "MySQL-to-PostgreSQL" using this tool

Did you migrate the contents as well, or only the table structure? 

From these steps, it sounds like you have a database that's fully up to date, but the rows in directus_migrations are missing. This in turn causes the warning to show up. Once the migrations are ran again, it'll fail because the alterations are already done. Make sure your original conversion from MySQL to Postgres contains both table schema _and_ data 👍🏻

**Reply by awnish-npg** [reply]
I have migrated the contents as well. I have also verified the directus_migration table (21 records)
![image](https://user-images.githubusercontent.com/60690093/128761003-7b4c64d5-7fd6-43e9-a3f2-3e71985fd4d8.png)
it has same number of records (MySQL and PostgreSQL).

**Reply by awnish-npg** [reply]
I also executed the command "directus bootstrap" which has now created the column "storage_default_folder" in directus_settings table but now throwing another error
 Applying Add Default Folder...
error: alter table "directus_settings" add column "storage_default_folder" uuid - column "storage_default_folder" of relation "directus_settings" already exists
![image](https://user-images.githubusercontent.com/60690093/128761258-24f9326d-4e9f-4303-a45b-74b9e944f3a1.png)
